sympy/polys/numberfields/galoisgroups.py

```diff
@@ -0,0 +1,808 @@
+            # is preferable to succeed with smaller coeffs.
+            # Give each coeff bound five tries, before incrementing.
+            C = min(i//5 + 1, max_coeff)
+            d = random.randint(2, n - 1)
+            a = dup_random(d, -C, C, ZZ)
+
+        A = Poly(a, T.gen)
+        U = Poly(T.resultant(X - A), X)
+        if U.rep not in history and dup_sqf_p(U.rep.rep, ZZ):
+            return A, U
+    raise MaxTriesException
+
+
+def M20():
+    """
+    Return a representation of the group M20, a subgroup of S5 that is one of
+    the possible Galois groups for polys of degree 5.
+    """
+    from sympy.combinatorics.perm_groups import PermutationGroup
+    from sympy.combinatorics.permutations import Permutation
+    return PermutationGroup(Permutation(0, 1, 2, 3, 4), Permutation(1, 2, 3, 4))
+
+
+def G36minus(verbose=False):
+    """
+    Return a representation of the group G36-, a subgroup of S6 isomorphic
+    to the semidirect product of C3^2 with C2^2.
+
+    This is one of the possible Galois groups for polys of degree 6 (not yet
+    supported!).
+    """
+    from sympy.combinatorics.perm_groups import PermutationGroup
+    from sympy.combinatorics.permutations import Permutation
+    # If verbose, conduct (and show) the search procedure whereby we found
+    # this representation initially. Otherwise just form and return the group.
+    if verbose:
+        from sympy.combinatorics.named_groups import SymmetricGroup
+        # For our normal subgroup N we take one obvious instance of C3^2 in S6:
+        N = PermutationGroup(
+            Permutation(5)(0, 1, 2),
+            Permutation(5)(3, 4, 5)
+        )
+        # Now we need an instance H of the Klein Viergruppe, such that
+        # H ^ N = {e}, and such that N will indeed be normal in N x H.
+        # This means finding two elements a != b in S6 such that:
+        #   order(a) == 2 == order(b)
+        #   ab = ba
+        #   a, b not in N
+        #   aNa^1 subseteq N supseteq bNb^1
+        cand = []
+        for a in SymmetricGroup(6).elements:
+            if a.order() == 2 and a not in N:
+                if all(a * n * (~a) in N for n in N.elements):
+                    cand.append(a)
+        print(f'\n{len(cand)} candidates')
+        print(cand)
+        for a in cand:
+            for b in cand:
+                if b == a:
+                    continue
+                c = a*b
+                if c != b*a:
+                    continue
+                if c not in cand:
+                    continue
+                print(f'{a} -- {b} -- {c}')
+    # Form and return the group.
+    G = PermutationGroup(
+        Permutation(5)(0, 1, 2),
+        Permutation(5)(3, 4, 5),
+        Permutation(5)(0, 1),
+        Permutation(5)(4, 5),
+    )
+    return G
+
+
+def is_square(n):
+    """Convenience to check if an Integer is a square. """
+    return isinstance(sqrt(n), Integer)
```

Comment:
```
from sympy.ntheory.primetest import is_square
```

Reply:
Thanks -- thought that ought to already exist somewhere!